include/host/wasi/vfs_io.h

```diff
@@ -0,0 +1,188 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2025 Second State INC
+
+#pragma once
+
+#include "host/wasi/environ.h"
+// #include "wasi/api.hpp"
+#include <cctype>
+#include <fstream>
+#include <ios>
+#include <sstream>
+#include <string>
+
+#if defined(_WIN32)
+#ifdef WASMEDGE_COMPILE_LIBRARY
+#define WASMEDGE_VFS_EXPORT __declspec(dllexport)
+#else
+#define WASMEDGE_VFS_EXPORT __declspec(dllimport)
+#endif
+#else
+#define WASMEDGE_VFS_EXPORT __attribute__((visibility("default")))
+#endif
+
+namespace WasmEdge {
+namespace FStream {
+
+enum FileFlags : uint16_t {
+  None = 0,
+  Exist = 1ULL << 0,
+  Read = 1ULL << 1,
+  Write = 1ULL << 2,
+};
+
+DEFINE_ENUM_OPERATORS(FileFlags)
+
+FileFlags getFileStats(Host::WASI::Environ *WASIEnv,
+                       const std::string_view &FileName) noexcept;
+
+bool fileExists(Host::WASI::Environ *WASIEnv,
+                const std::string_view &FileName) noexcept;
+
+bool canRead(Host::WASI::Environ *WASIEnv,
+             const std::string_view &FileName) noexcept;
+
+bool canWrite(Host::WASI::Environ *WASIEnv,
```

Comment:
I wonder these helper functions can be the member functions of `WASI::Environ` class.